src/scripts/main.js

```diff
@@ -1,3 +1,38 @@
 'use strict';
 
-// write your code here
+const table = document.querySelector('table');
+const tbody = table.querySelector('tbody');
+const thead = table.querySelector('thead');
+const tfoot = table.querySelector('tfoot');
+
+function cloneAndInsertColumn(indexToClone, insertBeforeIndex) {
+  // Клонування заголовка
+  const headerRow = thead.querySelector('tr');
+  const clonedHeaderCell = headerRow.children[indexToClone].cloneNode(true);
+
+  headerRow.insertBefore(
+    clonedHeaderCell,
+    headerRow.children[insertBeforeIndex],
+  );
+
+  const bodyRows = tbody.querySelectorAll('tr');
+
+  bodyRows.forEach((row) => {
+    const clonedCell = row.children[indexToClone].cloneNode(true);
```

Comment:
Make sure `indexToClone` is valid for each row in `bodyRows`. If rows have different numbers of cells, this could lead to errors.